lib/publiccloud/provider.pm

```diff
@@ -269,23 +307,45 @@ sub terraform_apply {
     my $image                = $self->get_image_id();
     my $ssh_private_key_file = '/root/.ssh/id_rsa';
     my $name                 = get_var('PUBLIC_CLOUD_RESOURCE_NAME', 'openqa-vm');
+    my $cloud_name           = $self->conv_openqa_tf_name;
 
     record_info('WARNING', 'Terraform apply has been run previously.') if ($self->terraform_applied);
 
-    assert_script_run('cd ' . TERRAFORM_DIR);
     record_info('INFO', "Creating instance $instance_type from $image ...");
+    if (get_var('PUBLIC_CLOUD_SLES4SAP')) {
+        assert_script_run('cd ' . TERRAFORM_DIR . "/$cloud_name");
+        my $sap_media   = get_required_var('MEDIA');
+        my $sap_regcode = get_required_var('SCC_REGCODE_SLES4SAP');
+        my $sle_version = get_required_var('VERSION');
+        $sle_version =~ s/-/_/g;
+        file_content_replace('terraform.tfvars',
+            '%MACHINE_TYPE%'         => $instance_type,
+            '%REGION%'               => $self->region,
```

Comment:
Hmmm this causes a weird tidy issue: https://travis-ci.org/os-autoinst/os-autoinst-distri-opensuse/jobs/595986667

Reply:
Hehe, I would guess it is a issue in travis, as the following will not work:
```
perl -w -e ' my $hash = { %MACHINE_TYPE% => "foo" };'
syntax error at -e line 1, near "% =>"
Execution of -e aborted due to compilation errors.
```

Reply:
But as the `%WHATEVER%` variables are protected with `' ... '` it should be OK, so yes more a Travis issue here I think (and code works, I tested it).

Reply:
What about using https://metacpan.org/pod/perlcritic#BENDING-THE-RULES
```
## no critic
...
## use critic
```

Reply:
Does perlcritic also complains if you do `q(%MACHINE_TYPE%)` instead of `'%MACHINE_TYPE%'`?

Reply:
Looking at https://github.com/os-autoinst/os-autoinst/blob/master/tools/lib/perlcritic/Perl/Critic/Policy/HashKeyQuotes.pm, `q(%MACHINE_TYPE%)` will fail the perlcritic check, but I don't think `q/%MACHINE_TYPE%/` will.

Reply:
Nice pointer. Didn't know that we maintain this HashKeyQuotes.pm thing by ourself. I was searching for it and wondered that it doesn't count in my local tests, thx!

I would guess there is an issue with https://github.com/os-autoinst/os-autoinst/blob/master/tools/lib/perlcritic/Perl/Critic/Policy/HashKeyQuotes.pm#L26 it should contain `%@`.

Reply:
@alvarocarvajald you where write `q/%REGION%/` will work as well. But I defiantly don't understand why! I added some debugs into `sub violates()` and using it, this method isn't called anymore.

All the others are already save, as they contain a `_` which is in that exception regex at line 26.

Reply:
Still added it here: https://github.com/os-autoinst/os-autoinst/pull/1223

Reply:
@cfconrad thx for taking care of this!

Reply:
Follow up to `q(%REGION%)` I think this should work as well. The explanation is:
https://metacpan.org/pod/PPI::Token::Quote::Single#SYNOPSIS

So to speed it up, we might just change it like this
```suggestion
            q(%REGION%)               => $self->region,
```